dev/gen_model_to_rdf.py

```diff
@@ -0,0 +1,334 @@
+        return "sha3_512"
+    return snake_case_to_camel_case(obj.name.lower())
+
+
+def model_to_rdf(obj, graph: Graph) -> Identifier:
+    if isinstance(obj, str):
+        return URIRef(obj)
+    type_name = obj.__class__.__qualname__
+    module_name = obj.__class__.__module__
+    namespace = module_to_namespace(module_name)
+    if namespace is None:
+        return literal_to_rdf(obj, graph)
+
+    lookup = f"{namespace}/{type_name}"
+    converter = _CONVERTER_FUNCTIONS[lookup] if lookup in _CONVERTER_FUNCTIONS else literal_to_rdf
+    return converter(obj, graph)
+"""
+
+FINAL_LINE = """# fmt: on
+"""
+
+output_dir = os.path.join(os.path.dirname(__file__), "../src/spdx_tools/spdx3/writer/rdf/converters")
+
+
+def prop_name_to_python(prop_name: str):
+    special_cases = {"from": "from_element", "homePage": "homepage"}
+    if prop_name in special_cases:
+        return special_cases[prop_name]
+    return camel_case_to_snake_case(prop_name)
+
+
+def namespace_name_to_python(namespace_name: str):
+    special_cases = {"AI": "ai"}
+    if namespace_name in special_cases:
+        return special_cases[namespace_name]
+    return camel_case_to_snake_case(namespace_name)
+
+
+class GenModelToRdf:
+    prop_name_to_id: dict[str, str]
+    class_to_converter_func: dict[str, str]
+    namespace_imports: str
+
+    def __init__(self):
+        self.prop_name_to_id = {}
+        self.class_to_converter_func = {}
+        self.namespace_imports = ""
+
+    def create_namespace_import(self, model: dict):
+        namespaces = [namespace_name_to_python(namespace["name"]) for namespace in model.values()]
+        if namespaces:
+            self.namespace_imports = "from . import " + ", ".join(namespaces)
+
+    def map_prop_names_to_ids(self, model: dict):
+        for namespace in model.values():
+            namespace_name = namespace["name"]
+            for prop_name, prop in namespace["properties"].items():
+                full_prop_name = f"{namespace_name}/{prop_name}"
+                prop_id = prop["metadata"]["id"]
+                self.prop_name_to_id[full_prop_name] = prop_id
+
+    def is_literal_type(self, typename: str, namespace_name: str, model: dict) -> bool:
+        if typename.startswith('/'):
+            typename = typename[1:]
+        if typename.startswith("xsd:"):
+            return True
+        if '/' in typename:
+            namespace_name, _, typename = typename.partition('/')
+        namespace = model[namespace_name] if namespace_name in model else None
+        if namespace and typename in namespace["vocabs"]:
+            return False
+        clazz = namespace["classes"][typename] if namespace and typename in namespace["classes"] else None
+        if not clazz:
+            return True
+        if "SubclassOf" not in clazz["metadata"] or clazz["metadata"]["SubclassOf"] == "none" or clazz["metadata"]["SubclassOf"].startswith("xsd:"):
+            return not clazz["properties"]
```

Comment:
Perhaps it would be sufficient to check for subclasses of xsd types without properties. A class without any parent and without any properties probably doesn't make sense in the spdx spec, anyway.